Extension/src/LanguageServer/renameDataProvider.ts

```diff
@@ -0,0 +1,154 @@
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+import * as vscode from 'vscode';
+import { RenameModel, RenamePendingFileItem, RenamePendingFilesGroupItem, RenamePendingItem,
+    RenameCandidateFileItem, RenameCandidateReferenceTypeGroupItem, RenameCandidateReferenceTypeItem,
+    RenameCandidateItem } from './renameModel';
+import { getReferenceTypeIconPath } from './referencesProvider';
+import { getReferenceTagString, ReferenceType } from './references';
+import * as nls from 'vscode-nls';
+
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+type TreeObject = RenamePendingFileItem | RenamePendingItem | RenameCandidateFileItem | RenameCandidateItem | RenameCandidateReferenceTypeGroupItem | RenameCandidateReferenceTypeItem;
+
+export class RenameDataProvider implements vscode.TreeDataProvider<TreeObject> {
+    private references: RenameModel;
+    private readonly _onDidChangeTreeData = new vscode.EventEmitter<TreeObject>();
+    readonly onDidChangeTreeData = this._onDidChangeTreeData.event;
+
+    constructor(readonly pending: boolean) {
+    }
+
+    setModel(renameModel: RenameModel): void {
+        this.references = renameModel;
+        this._onDidChangeTreeData.fire();
+    }
+
+    clear(): void {
+        this.references = undefined;
+        this._onDidChangeTreeData.fire();
+    }
+
+    update(): void {
+        this._onDidChangeTreeData.fire();
+    }
+
+    getTreeItem(element: TreeObject): vscode.TreeItem {
+        if (!this.references) {
+            return;
+        }
+
+        if (element instanceof RenamePendingItem) {
+            const result: vscode.TreeItem = new vscode.TreeItem(element.text);
+            result.collapsibleState = vscode.TreeItemCollapsibleState.None;
+            result.iconPath = getReferenceTypeIconPath(element.type);
+            let tag: string = getReferenceTagString(element.type, false);
+            result.tooltip = `[${tag}]\n${element.text}`;
+            result.command = {
+                title: localize("goto.reference", "Go to reference"),
+                command: 'C_Cpp.ShowReferenceItem',
+                arguments: [element]
+            };
+            result.contextValue = "pendingItem";
+            return result;
+        }
+
+        if (element instanceof RenamePendingFileItem) {
+            const result: vscode.TreeItem = new vscode.TreeItem(element.uri);
+            result.collapsibleState = vscode.TreeItemCollapsibleState.Expanded;
+            result.iconPath = vscode.ThemeIcon.File;
+            result.description = true;
+            result.contextValue = "pendingFile";
+            let tag: string = getReferenceTagString(ReferenceType.ConfirmationInProgress, false);
```

Comment:
Pass in element.type instead of "ReferenceType.ConfirmationInProgress"?

Reply:
This is actually the file level.  I removed the tooltip for it.